### PR TITLE
Version utility class for gcode comment and title in dev

### PR DESCRIPTION
**Pull request by PPAC37**
The details of the version in addition if DEV=true. ( JFrame title and .gcode Header)

### CODE REVIEW

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java [hunk 1]
```diff
@@ -56,17 +59,38 @@ private String addExtension(String name, String [] extensions) {
...
+			}
+			
 			SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
 			Date date = new Date(System.currentTimeMillis());
 			out.write("; " + formatter.format(date) + "\n");
+			
 			// TODO MarlinPlotterInterface.getFindHomeString()?
 			out.write("G28\n");  // go home
 
+			// TODO user "start gcode"
```
Comment (coliss86) [inline src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java:92]: If your PR is ready to be merged, there should be no TODO. If you want to keep track of something else to do, please create an issue.

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java [hunk 2]
```diff
@@ -56,17 +59,38 @@ private String addExtension(String name, String [] extensions) {
...
 	}
 	
+	boolean detailGCodeHeader = true;
+	
 	private void save(String filename, Turtle turtle, Plotter robot) throws Exception {
 		logger.debug("saving...");
 		
 		try (Writer out = new OutputStreamWriter(new FileOutputStream(filename))) {
+			
+			if (detailGCodeHeader) {
```
Comment (coliss86) [inline src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java:69]: when this boolean is set to `false` ?

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java [hunk 3]
```diff
@@ -56,17 +59,38 @@ private String addExtension(String name, String [] extensions) {
...
+				out.write(";FLAVOR:Marlin-polargraph\n");
+
+				//out.write(";TIME:"+ "\n"); // TODO
+				
+				Rectangle2D.Double bounds = turtle.getBounds();
```
Comment (coliss86) [inline src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java:74]: it is an excellent idea but it is not related to the title of the PR. One PR = one subject and only one.

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java (hunk 1 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java]: Issue created. thank you.
But I would have kept the TODO in the code too ... (double reminder )

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java (hunk 2 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java]: I have the good or bad habit of making blocks subject to a decarred boolean as a class attribute ... easier to deactivate when recoding / less clean for a production version ...

File: src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java (hunk 3 above)
Comment (PPAC37) [reply src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java]: So a new PR to come :)

Comment (coliss86) [reply src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java]: 👍

**Review by i-make-robots — APPROVED** [review]

**Pull request merged**